doc/coq.rst

```diff
@@ -0,0 +1,379 @@
+.. _coq:
+
+***
+Coq
+***
+
+.. contents:: Table of Contents
+    :depth: 3
+
+Introduction
+------------
+
+Dune is able to build Coq theories and plugins with support for source file
+preprocessing, and extraction.
+
+A *Coq theory* is a collection of ``.v`` files, perhaps in several directories,
+all prefixed by a common Coq module name, following the directory hierarchy for
```

Comment:
Be more explicit.

Reply:
How is this too vague or broad? What else can a Coq theory be? Even if the .v files are generated, they are still there.

Reply:
The generated part seems redundant, but that has been a long time FAQ so IMHO it can't hurt to clarity that here for people not familiar with dune.

Reply:
Who is generating .v files?

Reply:
Off the top of my head BedRock people, most MIT projects, OTT users, and likely I miss some.

Reply:
Right, but I am unaware that any of these projects use `dune`'s `coq.theory`.

Reply:
Why complicate the text here?

Reply:
You are unaware but at least some of them (private) use it. It is not easy indeed to write this text, maybe we can add instead a separate sentence at the end. "Note that .v files, as usual in Dune, can originate from the source tree, or be generated by a user rule"

Reply:
The reason I ask this is that people coming from make fail to consistently understand that dune will do a glob over the full set of targets, instead of the filesystem users can see.